Added check_assumptions in Main Namespace

arihantparsoya opened the pull request:
This is Fix for #11505. Added `check_assumptions()` in main namespace by adding it in _init_.py.
#### Sample Program:

``` python
>>> from sympy import *
>>> check_assumptions(-5, integer=True)
True
```

arihantparsoya committed:
added check_assumptions in main namespace

Added check_assumptions() in main namespace by adding it in _init_.py.

arihantparsoya committed:
Implemented check_assumptions to check values against expressions

Fix for #11505. Previously, function check_function could only accept
assumptions in form of dict.

By adding another input variable `e` the function can now accept
variable or expression to compare with `expr`.

If e is not none, the `assumptions` will be set to `e.assumptions0`.

parsoyaarihant commented:
@asmeurer , can you review this?

asmeurer commented on sympy/solvers/solvers.py:
```diff
@@ -331,6 +331,9 @@ def check_assumptions(expr, **assumptions):
        >>> z = Symbol('z')
        >>> check_assumptions(z, real=True)
     """
+    if e is not None:
```
Can you document this option in the docstring?

asmeurer replied:
Maybe a better name for this argument would be `against` (i.e., you are checking the assumptions of `expr` *against* the assumptions of `against`).

arihantparsoya committed:
Updated Docstring and Changed Argument Name

Changed the argument name `e` to `against` for better core readability.

Added documentation of new functionality in the docstring.

Added tests.

parsoyaarihant commented:
@asmeurer , I have updated the docstrings.

asmeurer commented on sympy/solvers/solvers.py:
```diff
@@ -325,12 +324,21 @@ def check_assumptions(expr, **assumptions):
        False
 
+       To check assumptions of `expr` against another variable or expression,
+       pass the expression or variabls as `against`.
```
Spelling

parsoyaarihant replied:
Thanks, required changes are committed.

asmeurer commented on sympy/solvers/solvers.py:
```diff
@@ -325,12 +324,21 @@ def check_assumptions(expr, **assumptions):
        False
 
+       To check assumptions of `expr` against another variable or expression,
+       pass the expression or variable as `against`.
```
Use two backquotes in docstrings.

parsoyaarihant replied:
Thanks, changes are committed.
